job/replay_manager.go

```diff
@@ -90,6 +109,117 @@ func (m *Manager) Replay(reqInput *models.ReplayWorkerRequest) (string, error) {
+		if err != nil {
+			return err
+		}
+
+		//check another replay active for this dag
+		activeReplaySpecs, err := replaySpecRepo.GetByStatus(ReplayStatusToValidate)
+		if err != nil {
+			if err == store.ErrResourceNotFound {
+				return nil
+			}
+			return err
+		}
+		return validateReplayJobsConflict(activeReplaySpecs, reqInput, reqReplayNodes)
+	}
+	//check and cancel if found conflicted replays for same job ID
+	return cancelConflictedReplays(replaySpecRepo, reqInput)
+}
+
+func cancelConflictedReplays(replaySpecRepo store.ReplaySpecRepository, reqInput *models.ReplayWorkerRequest) error {
+	duplicatedReplaySpecs, err := replaySpecRepo.GetByJobIDAndStatus(reqInput.Job.ID, ReplayStatusToValidate)
+	if err != nil {
+		if err == store.ErrResourceNotFound {
+			return nil
+		}
+		return err
+	}
+	for _, replaySpec := range duplicatedReplaySpecs {
+		if err := replaySpecRepo.UpdateStatus(replaySpec.ID, models.ReplayStatusCancelled, models.ReplayMessage{
+			Type:    ErrConflictedJobRun.Error(),
+			Message: fmt.Sprintf("force started replay with ID: %s", reqInput.ID),
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *Manager) validateRunningInstance(ctx context.Context, reqReplayNodes []*tree.TreeNode, reqInput *models.ReplayWorkerRequest) error {
+	requestBatchSize := 100
+	for _, reqReplayNode := range reqReplayNodes {
+		batchEndDate := reqInput.End.AddDate(0, 0, 1)
+		jobStatusAllRuns, err := m.scheduler.GetDagRunStatus(ctx, reqInput.Project, reqInput.Job.Name, reqInput.Start, batchEndDate, requestBatchSize)
+		if err != nil {
+			return err
+		}
+		for _, jobStatus := range jobStatusAllRuns {
+			if reqReplayNode.Runs.Contains(jobStatus.ScheduledAt) && jobStatus.State == models.JobStatusStateRunning {
+				return ErrConflictedJobRun
+			}
+		}
+	}
+	return nil
+}
+
+func validateReplayJobsConflict(activeReplaySpecs []models.ReplaySpec, reqInput *models.ReplayWorkerRequest,
+	reqReplayNodes []*tree.TreeNode) error {
+	for _, activeSpec := range activeReplaySpecs {
+		activeReplayWorkerRequest := &models.ReplayWorkerRequest{
+			ID:         activeSpec.ID,
+			Job:        activeSpec.Job,
+			Start:      activeSpec.StartDate,
+			End:        activeSpec.EndDate,
+			Project:    reqInput.Project,
+			JobSpecMap: reqInput.JobSpecMap,
+		}
+		activeTree, err := prepareTree(activeReplayWorkerRequest)
+		if err != nil {
+			return err
+		}
+		activeNodes := activeTree.GetAllNodes()
+
+		return checkAnyConflictedDags(activeNodes, reqReplayNodes)
+	}
+	return nil
+}
+
+func checkAnyConflictedDags(activeNodes []*tree.TreeNode, reqReplayNodes []*tree.TreeNode) error {
+	activeNodesMap := make(map[string]*tree.TreeNode)
+	for _, activeNodes := range activeNodes {
+		activeNodesMap[activeNodes.GetName()] = activeNodes
+	}
+
+	for _, reqNode := range reqReplayNodes {
+		if activeNodesMap[reqNode.Data.GetName()] != nil {
+			if err := checkAnyConflictedRuns(activeNodesMap[reqNode.Data.GetName()], reqNode); err != nil {
```

Comment:
@mauliksoneji added if logic here, so it wont returning in the first loop

Reply:
check if the value is present in map by using
```
if val, ok := dict["foo"]; ok {
    //do something here
}
```

Reply:
use GetName() instead of Data.GetName()